api/gateway.go

```diff
@@ -41,6 +45,10 @@ func (srv *Server) gatewayAddHandler(w http.ResponseWriter, req *http.Request, p
 // gatewayRemoveHandler handles the API call to remove a peer from the gateway.
 func (srv *Server) gatewayRemoveHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 	addr := modules.NetAddress(ps.ByName("netaddress"))
+	if err := addr.IsValid(); err != nil {
+		writeErorr(w, "invalid address: "+err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
Validation is not required in many places. It is required only when an address is going to be stored previously being dialed. AFAIK, that only occurs in two places: in creating/reading host announcements, and in the gateway's node sharing.
Regardless, I don't think validation should be performed in the API package. The API package strives to be a lightweight wrapper around the module objects. It should introduce as little additional logic as possible.

>As a side note if you pass an un-dialable NetAddress to Connect it will add it to it's peer list before it returns an error.

I don't think this is the case. Maybe you were confused by these lines:
```go
	id := g.mu.RLock()
	_, exists := g.peers[addr]
	g.mu.RUnlock(id)
	if exists {
		return errors.New("peer already added")
	}
```